### DISCUSSION TITLE
Proper nginx configuration for WEB_DOMAIN

### DISCUSSION THREAD

**Discussion by mieth-robert** [mastodon/mastodon#18740] [Q&A] [answered]
Hi All! 
I am a newbie in running a server, but so far everything works swimmingly. Now I ran into a problem that I can not figure out. Instead of using my domain example.com I would like to access the server via social.example.com. I set the .env.production file as described [here](https://docs.joinmastodon.org/admin/config/) and added `location /.well-known/webfinger { return 301 https://social.example.com$request_uri; }` to the (https) server block of example.com. But when I try to access social.example.com I receive a privacy error. I tried various nginx configurations but none worked. What could I have missed? I was not able to find more detailed documentation somewhere else. 

Hers is the relevant part of my nginx configuration (/etc/nginx/sites-available/mastodon): 

```
[...]

server {
  listen 80;
  listen [::]:80;
  server_name example.com;
  root /home/mastodon/live/public;
  location /.well-known/acme-challenge/ { allow all; }
  location / { return 301 https://$host$request_uri; }

}

server {
  listen 443 ssl http2;
  listen [::]:443 ssl http2;
  server_name example.com;

  location /.well-known/webfinger { return 301 https://social.example.com$request_uri; }

  ssl_protocols TLSv1.2 TLSv1.3;
  ssl_ciphers HIGH:!MEDIUM:!LOW:!aNULL:!NULL:!SHA;
  ssl_prefer_server_ciphers on;
  ssl_session_cache shared:SSL:10m;
  ssl_session_tickets off;

  # Uncomment these lines once you acquire a certificate:
  ssl_certificate     /etc/letsencrypt/live/example.com/fullchain.pem;
  ssl_certificate_key /etc/letsencrypt/live/example.com/privkey.pem;

  keepalive_timeout    70;
  sendfile             on;
  client_max_body_size 80m;

[...]

}

[...]
```

**Comment by ClearlyClaire** [comment, marked as answer]
A privacy error? Have you checked that your certificate covers your subdomain?

On a side note, it sounds like you had your instance already running at `example.com` before moving the `WEB_DOMAIN` to `social.example.com`. As the documentation says, this is likely to cause issues with federating with instances that already interacted with you.

**Reply by mieth-robert** [reply]
That was it! I knew I missed something simple. 
All settings are as described above and I ran

`certbot certonly -d example.com,social.example.com`

(no space after comma!) and chose 1: Nginx, and (E)xpand in the following prompts. Then restarted Nginx and Mastodon and it works now. 

Thanks for the warning, but everything was fresh and I don't think any instance has interacted with me yet.

Cheers!
